### PR TITLE
Feature/add convnext v2

### PR DESCRIPTION
**Pull request by learncat163**
## 概述

本PR为PaddleClas添加了ConvNeXt-V2模型系列的支持，包含8个不同规模的模型变体。权重从timm (PyTorch) 转换而来，已通过严格的精度对齐验证：单张推理差异 < 2.06e-05，ImageNet Top-1 Acc 误差 0.00%。

---

### 支持的模型

| 模型名称 | 参数量 | conv_mlp | depths | dims | ImageNet Top-1 |
|---------|--------|----------|--------|------|---------------|
| ConvNeXtV2_atto | 3.7M | True | [2,2,6,2] | [40,80,160,320] | 76.65% |
| ConvNeXtV2_femto | 5.0M | True | [2,2,6,2] | [48,96,192,384] | 78.48% |
| ConvNeXtV2_pico | 7.0M | True | [2,2,6,2] | [64,128,256,512] | 80.31% |
| ConvNeXtV2_nano | 12.0M | True | [2,2,8,2] | [80,160,320,640] | 81.87% |
| ConvNeXtV2_tiny | 28.0M | False | [3,3,9,3] | [96,192,384,768] | 82.94% |
| ConvNeXtV2_base | 88.0M | False | [3,3,27,3] | [128,256,512,1024] | 84.88% |
| ConvNeXtV2_large | 195.0M | False | [3,3,27,3] | [192,384,768,1536] | 85.76% |
| ConvNeXtV2_huge | 660.0M | False | [3,3,27,3] | [352,704,1408,2816] | 86.25% |

---

### logits Diff对齐

| 模型    | Max Abs Diff |
|-------|--------------|
| atto  | 3.58e-06     |
| femto | 1.67e-06     |
| pico  | 2.62e-06     |
| nano  | 2.04e-05     |
| tiny  | 1.67e-06     |
| base  | 6.20e-06     |
| large | 2.86e-06     |
| huge  | 5.48e-06     |

<details>
<summary>验证脚本: diff_inference_compare.py (点击展开)</summary>

```python
import numpy as np
import torch
import paddle
import timm
from scipy.spatial.distance import cosine


def compare_single_image(model_name: str):
    variant = model_name.replace("timm/", "").replace(".fcmae_ft_in1k", "")

    np.random.seed(42)
    x_np = np.random.randn(1, 3, 224, 224).astype(np.float32)

    mean = np.array([0.485, 0.456, 0.406], dtype=np.float32).reshape(1, 3, 1, 1)
    std = np.array([0.229, 0.224, 0.225], dtype=np.float32).reshape(1, 3, 1, 1)
    x_np = ((x_np - mean) / std).astype(np.float32)

    # timm: 官方预训练权重
    timm_model = timm.create_model(model_name, pretrained=True, num_classes=1000)
    timm_model.eval()
    with torch.no_grad():
        out_timm = timm_model(torch.from_numpy(x_np).float()).cpu().numpy()

    # Paddle: 对应ConvNeXtV2模型推理
    from ppcls.arch.backbone.model_zoo.convnext import ConvNeXtV2
    model_configs = {
        "atto": {"depths": [2, 2, 6, 2], "dims": [40, 80, 160, 320]},
        "femto": {"depths": [2, 2, 6, 2], "dims": [48, 96, 192, 384]},
        "pico": {"depths": [2, 2, 6, 2], "dims": [64, 128, 256, 512]},
        "nano": {"depths": [2, 2, 8, 2], "dims": [80, 160, 320, 640]},
        "tiny": {"depths": [3, 3, 9, 3], "dims": [96, 192, 384, 768]},
        "base": {"depths": [3, 3, 27, 3], "dims": [128, 256, 512, 1024]},
        "large": {"depths": [3, 3, 27, 3], "dims": [192, 384, 768, 1536]},
        "huge": {"depths": [3, 3, 27, 3], "dims": [352, 704, 1408, 2816]},
    }
    cfg = model_configs[variant]
    paddle_model = ConvNeXtV2(
        in_chans=3, class_num=1000, depths=cfg["depths"], dims=cfg["dims"], use_grn=True
    )
    paddle_model.eval()
    with paddle.no_grad():
        out_paddle = paddle_model(paddle.to_tensor(x_np)).numpy()

    # Diff指标
    abs_diff = np.abs(out_timm - out_paddle)
    max_abs_diff = abs_diff.max()
    mean_abs_diff = abs_diff.mean()
    cosine_sim = 1 - cosine(out_timm.flatten(), out_paddle.flatten())
    passed = max_abs_diff < 1e-4

    return {
        "max_abs_diff": float(max_abs_diff),
        "mean_abs_diff": float(mean_abs_diff),
        "cosine_similarity": float(cosine_sim),
        "passed": bool(passed),
    }
```

</details>

### ImageNet Top-1 Accuracy 对齐 (50000张验证集)

| 模型    | timm Top-1 | Paddle Top-1 | 误差    |
|-------|------------|--------------|-------|
| atto  | 76.65%     | 76.65%       | 0.00% |
| femto | 78.48%     | 78.48%       | 0.00% |
| pico  | 80.31%     | 80.31%       | 0.00% |
| nano  | 81.87%     | 81.87%       | 0.00% |
| tiny  | 82.94%     | 82.94%       | 0.00% |
| base  | 84.88%     | 84.88%       | 0.00% |
| large | 85.76%     | 85.76%       | 0.00% |
| huge  | 86.25%     | 86.25%       | 0.00% |

<details>
<summary>验证脚本: topacc_validate.py (点击展开)</summary>

```python
import numpy as np

INPUT_SIZE = 224
IMAGENET_DEFAULT_MEAN = (0.485, 0.456, 0.406)
IMAGENET_DEFAULT_STD = (0.229, 0.224, 0.225)


def load_imagenet_val_dataset(imagenet_dir):
    # synset 目录按字母排序映射 class id
    synset_dirs = sorted([d for d in imagenet_dir.iterdir() if d.is_dir() and d.name.startswith('n')])
    synset_to_classid = {d.name: idx for idx, d in enumerate(synset_dirs)}
    all_image_paths, all_gt_labels = [], []
    for d in synset_dirs:
        cid = synset_to_classid[d.name]
        for img_path in sorted(d.glob('*.JPEG')):
            all_image_paths.append(img_path)
            all_gt_labels.append(cid)
    return all_image_paths, all_gt_labels


def load_timm_model(model_name):
    import torch
    import timm
    timm_model = timm.create_model(model_name, pretrained=True, num_classes=1000)
    timm_model = timm_model.to('cuda').eval()
    return timm_model, 'cuda'


def load_paddle_model(model_name):
    from ppcls.arch.backbone.model_zoo.convnext import (
        ConvNeXtV2_atto, ConvNeXtV2_femto, ConvNeXtV2_pico, ConvNeXtV2_nano,
        ConvNeXtV2_tiny, ConvNeXtV2_base, ConvNeXtV2_large, ConvNeXtV2_huge,
    )
    paddle_builders = {
        'atto': ConvNeXtV2_atto,
        'femto': ConvNeXtV2_femto,
        'pico': ConvNeXtV2_pico,
        'nano': ConvNeXtV2_nano,
        'tiny': ConvNeXtV2_tiny,
        'base': ConvNeXtV2_base,
        'large': ConvNeXtV2_large,
        'huge': ConvNeXtV2_huge,
    }
    model = paddle_builders[model_name](pretrained=True)
    model.eval()
    return model


def compute_top1(predictions, gt_labels):
    gt_array = np.array(gt_labels, dtype=np.int32)
    top1_correct = int(np.sum(predictions == gt_array))
    return top1_correct / len(gt_labels) * 100


def judge_alignment(paddle_acc, timm_acc):
    diff = abs(paddle_acc - timm_acc)
    return diff, (diff < 0.2)
```

</details>

---

## 使用示例

```python
import paddle
from ppcls.arch.backbone.model_zoo import ConvNeXtV2_tiny

# 创建模型 (自动下载预训练权重)
model = ConvNeXtV2_tiny(pretrained=True)

# 推理
x = paddle.randn([1, 3, 224, 224])
output = model(x)
pred = paddle.argmax(output, axis=1)
print(f"Predicted class: {pred.item()}")
```

---

### CODE REVIEW

**Comment by zhangyubo0722** [comment]
缺少训练配置文件

**Comment by learncat163** [comment]
> 缺少训练配置文件

已经添加

**Review by TingquanGao — APPROVED** [review]
